src/main/java/org/terasology/lost/OnSpawnSystem.java

```diff
@@ -15,26 +15,139 @@
     @In
     private EntityManager entityManager;
     @In
     private InventoryManager inventoryManager;
+    @In
+    private Console console;
+    @In
+    private AssetManager assetManager;
+    @In
+    private WorldProvider worldProvider;
 
     @ReceiveEvent
     public void onPlayerSpawn(OnPlayerSpawnedEvent event, EntityRef player, InventoryComponent inventory) {
         inventoryManager.giveItem(player, null, entityManager.create("Lost:antrumSabre"));
+        ProgressTrackingComponent progressTrackingComponent = new ProgressTrackingComponent();
+        progressTrackingComponent.addLevel("Grassland", "Lost:well");
+        progressTrackingComponent.addLevel("Tundra", "Lost:well");
+        progressTrackingComponent.addLevel("Bare", "Lost:well");
+        progressTrackingComponent.addLevel("Scorched", "Lost:well");
+        progressTrackingComponent.addLevel("Taiga", "Lost:well");
+        progressTrackingComponent.addLevel("Temperate desert", "Lost:well");
+        progressTrackingComponent.addLevel("Temperate rain forest", "Lost:well");
+        progressTrackingComponent.addLevel("Temperate deciduous forest", "Lost:well");
+        progressTrackingComponent.addLevel("Subtropical desert", "Lost:well");
+        progressTrackingComponent.addLevel("Shrubland", "Lost:well");
+        progressTrackingComponent.addLevel("Marsh", "Lost:well");
+        progressTrackingComponent.addLevel("Tropical rain forest", "Lost:well");
+        progressTrackingComponent.addLevel("Tropical seasonal forest", "Lost:well");
+        progressTrackingComponent.addLevel("Lakeshore", "Lost:well");
+        progressTrackingComponent.addLevel("Coast", "Lost:well");
+        progressTrackingComponent.addLevel("Ice", "Lost:well");
+        player.addComponent(progressTrackingComponent);
+    }
+
+    @ReceiveEvent
+    public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player) {
+        console.addMessage("OLD:" + event.getOldBiome().getDisplayName());
+        console.addMessage("NEW:" + event.getNewBiome().getDisplayName());
+        LocationComponent loc = player.getComponent(LocationComponent.class);
+        Vector3f pos = loc.getWorldPosition();
+        ProgressTrackingComponent progressTrackingComponent = player.getComponent(ProgressTrackingComponent.class);
+        int searchRadius = 0;
+        Vector3i ext = new Vector3i(searchRadius, 0, searchRadius);
+        Vector3i desiredPos = new Vector3i(pos.getX(), 1, pos.getZ());
+
+        // try and find somewhere in this region a spot to land
+        Region3i spawnArea = Region3i.createFromCenterExtents(desiredPos, ext);
+        Region worldRegion = LostWorldGenerator.world.getWorldData(spawnArea);
+
+        GraphFacet graphs = worldRegion.getFacet(GraphFacet.class);
+        WhittakerBiomeModelFacet model = worldRegion.getFacet(WhittakerBiomeModelFacet.class);
+        Vector2f pos2d = new Vector2f(pos.getX(), pos.getZ());
+        double d = 1000000;
```

Comment:
Avoid "magic numbers" if able - if something is a meaningful number you could make it a named constant then assign it to a variable named something shorter. Makes it easier to understand what's going on. Even without a constant having a nicely named variable can go a long way :-)